### ISSUE TITLE
SequenceClassifier doesn't seem to learn leaked target

### ISSUE THREAD

**Issue by Ingvar-Y** [OpenNMT/OpenNMT-tf#957]
I tried a simple test with SequenceRecordInputter and LSTMEncoder where I created random binary target and then leaked it to train data with some noise:
```python3
import numpy as np
from opennmt import encoders, inputters, models, Runner
rng = np.random.default_rng(2022)
shapes = rng.integers(1, 9, 100000)
rands = []
for i in shapes:
    rands.append(rng.random((i, 16)))
target = rng.integers(0, 2, 100000)
for i in range(100000):
    rands[i][:, 0] += np.arange(rands[i].shape[0]) * 0.1 + target[i]
inputters.create_sequence_records(rands, "tgt_with_noise.records")
with open("tgt_labels.txt", "w") as f:
    f.write("\n".join(target.astype("str")))
with open("tgt_vocab.txt", "w") as f:
    f.write("\n".join(["0", "1"]))
config = {
    "model_dir": ".",
    "data": {
        "target_vocabulary": "tgt_vocab.txt",
        "train_features_file": "tgt_with_noise.records",
        "train_labels_file": "tgt_labels.txt",
    },
    "params": {"optimizer": "Adam", "learning_rate": 0.001},
    "train": {
        "batch_size": 8,
        "max_step": 5000,
        "sample_buffer_size": -1,
        "save_checkpoints_steps": 1000,
    },
}
runner = Runner(model, config, auto_config=False)
runner.train()
runner.infer(
    "tgt_with_noise.records",
    checkpoint_path="ckpt-5000.index",
    predictions_file="train_pred.txt",
)
test_values = np.loadtxt("tgt_labels.txt")
pred_values = np.loadtxt("train_pred.txt")
(test_values == pred_values).sum() / len(test_values)
```
As a result I got accuracy ~50% meaning the model doesn't work any better than a random guess. It seems 5000 steps should be enough for a model to get basic correspondence, what seems to be problem here?

**Comment by guillaumekln** [comment]
The `checkpoint_path` argument passed to `infer` is incorrect and so it is running inference with non trained weights. `checkpoint_path` should be a path to a checkpoint directory or a checkpoint prefix such as `ckpt-5000` (without `.index`).

You can also remove this argument and the latest checkpoint will be loaded automatically.

**Comment by Ingvar-Y** [comment]
I see, that solves the issue, thank you
